File: E38-count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        s = '1'
        for _ in range(n-1):
            l, ss, count = s[0], '', 0
            for i in s:
                if l == i:
                    count += 1
                else:
                    ss += str(count)+l
                    l = i
                    count = 1
            ss += str(count)+l
            s = ss
        return s

File: Java/E38-count-and-say.java
class Solution {
    public String countAndSay(int n) {
        String res = "1$";
        while(n > 1) {
            int count = 0;
            Character pre = res.charAt(0);
            String nextres = "";
            for (Character c:res.toCharArray()) {
                if (pre == c) {
                    count ++;
                } else {                    
                    nextres += String.valueOf(count) + String.valueOf(pre);
                    pre = c;
                    count = 1;
                }
            }
            res = nextres + "$";
            n --;
        }
        return res.substring(0, res.length() - 1);
    }
}